src/modules/ekf2/EKF/mag_heading_control.cpp

```diff
@@ -68,18 +68,34 @@ void Ekf::controlMagHeadingFusion(const magSample &mag_sample, const bool common
 	if (_control_status.flags.yaw_align) {
 		// mag heading
 		aid_src.innovation = wrap_pi(getEulerYaw(_R_to_earth) - aid_src.observation);
+		_mag_heading_innov_lpf.update(aid_src.innovation);
+
+		if (!_control_status.flags.gps) {
```

Comment:
I agree with this logically, but could we do a better job conveying the intention (importance of minimal mag heading innovations while GPS active) in a slightly different form?

``` C++
if (_control_status.flags.gps) {
    // consider mag consistent if filtered innovations are small once we have GPS
    if (fabsf(_mag_heading_innov_lpf.getState()) < math::radians(10.f)) {
	    // yaw angle must be observable to consider consistency
	    if (_yaw_angle_observable) {
		    _control_status.flags.mag_consistent = true;
	    }
    
    } else {
         _control_status.flags.mag_consistent = false;
    }

} else {
    // by default consider mag consistent if GPS unavailable
    _control_status.flags.mag_consistent = true;
}


```